tests/test_notls/test_notls.c

```diff
@@ -37,6 +37,8 @@ int solo5_app_main(const struct solo5_start_info *si __attribute__((unused)))
                        "add x0, x0, #0x10; "
                        "ldr w1, [x0]"
                        : : : "x0", "w1");
+#elif defined(__powerpc64__)
+    __asm__ __volatile("ld 3,-28672(13)" : : : "r3", "r13");
```

Comment:
The magic index of -28672 presumably comes from compiler generated code? Is that the default offset of a TCB on PowerPC?

Reply:
Have a look at Figure 3.1 here: http://openpowerfoundation.org/wp-content/uploads/resources/leabi/content/dbdoclet.50655241_50444.html

The TP is in register r13. It accesses the first TLS variable using -0x7000(r13). They did this because of efficiency purposes and due to addressing modes supported by the processor where that offset is 16bit signed.